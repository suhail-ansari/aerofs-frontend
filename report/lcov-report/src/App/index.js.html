<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/App/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/App</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>15/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>15/16</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import ReactDOM from 'react-dom';
&nbsp;
import MessageInput from '../MessageInput';
&nbsp;
export default class App extends Component {
  constructor() {
    super();
&nbsp;
    this.state = {
      error: '',
      text: '',
      messages: [],
      last_seen: 0
    };
&nbsp;
    this.sendHandler = this.sendMessage.bind(
      this,
      { key: 'Enter' }
    );
    this.sendMessage = this.sendMessage.bind(this);
    this.updateText = this.updateText.bind(this);
    this._updateMessages = this._updateMessages.bind(this);
    this._handleError = this._handleError.bind(this);
&nbsp;
    this.getMessages();
  }
&nbsp;
  getMessages() {
    fetch('/fixtures/fakedata.json')
      .then(res =&gt; res.json())
      .then(this._updateMessages)
      .catch(this._handleError);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  _u</span>pdateMessages(data) {
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      ...this.state,
      messages: [
        ...data.messages,
        ...this.state.messages
      ]
    });
  }
&nbsp;
  _handleError(err) {
    this.setState({
      ...this.state,
      error: 'There was an error while fetching messages.'
    });
  }
&nbsp;
  updateText({ target }) {
    this.setState({
      ...this.state,
      text: target.value
    });
  }
&nbsp;
  sendMessage({ key }) {
    <span class="missing-if-branch" title="else path not taken" >E</span>if (key == 'Enter' &amp;&amp; this.state.text !== '') {
      console.log(this.state);
    }
  }
&nbsp;
  render() {
    return (
      &lt;MessageInput
        text={this.state.text}
        changeHandler={this.updateText}
        keyUpHandler={this.sendMessage}
        sendHandler={this.sendHandler} /&gt;
    );
  }
&nbsp;
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 04 2017 01:48:16 GMT-0700 (PDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
