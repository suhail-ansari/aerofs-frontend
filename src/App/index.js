import React, { Component } from 'react';
import ReactDOM from 'react-dom';

import MessageInput from '../MessageInput';

export default class App extends Component {
  constructor() {
    super();

    this.state = {
      error: '',
      text: '',
      messages: [],
      last_seen: 0
    };

    this.sendHandler = this.sendMessage.bind(
      this,
      { key: 'Enter' }
    );
    this.sendMessage = this.sendMessage.bind(this);
    this.updateText = this.updateText.bind(this);
    this._updateMessages = this._updateMessages.bind(this);
    this._handleError = this._handleError.bind(this);

    this.getMessages();
  }

  getMessages() {
    fetch('/fixtures/fakedata.json')
      .then(res => res.json())
      .then(this._updateMessages)
      .catch(this._handleError);
  }

  _updateMessages(data) {
    this.setState({
      ...this.state,
      messages: [
        ...data.messages,
        ...this.state.messages
      ]
    });
  }

  _handleError(err) {
    this.setState({
      ...this.state,
      error: 'There was an error while fetching messages.'
    });
  }

  updateText({ target }) {
    this.setState({
      ...this.state,
      text: target.value
    });
  }

  sendMessage({ key }) {
    if (key == 'Enter' && this.state.text !== '') {
      console.log(this.state);
    }
  }

  render() {
    return (
      <MessageInput
        text={this.state.text}
        changeHandler={this.updateText}
        keyUpHandler={this.sendMessage}
        sendHandler={this.sendHandler} />
    );
  }

}