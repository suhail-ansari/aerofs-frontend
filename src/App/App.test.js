import React from 'react';
import renderer from 'react-test-renderer';
import App from './index';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));

test('should initialize', () => {
  let tree = renderer.create(
    <App/>
  );
  expect(tree).toMatchSnapshot();
});