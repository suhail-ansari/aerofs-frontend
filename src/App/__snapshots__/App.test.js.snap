// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should initialize 1`] = `
<div>
  <input
    onChange={[Function]}
    onKeyUp={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    Send
  </button>
</div>
`;
